src/Compilers/Core/Portable/SpecialTypeExtensions.cs

```diff
@@ -57,6 +57,35 @@ public static bool IsBlittable(this SpecialType specialType)
             }
         }
 
+        /// <summary>
+        /// Checks if a type is an immutable value type primitive.
+        /// </summary>
+        public static bool IsImmutableValueTypePrimitive(this SpecialType specialType)
```

Comment:
Are we really trying to "defend" against the possibility that we'll add a platform special value type that has a mutating ToString() method? That seems really, really unlikely. I'd rather see this method removed and have the simpler test in the caller.